src/borg/archive.py

```diff
@@ -723,15 +725,11 @@ class ChunksIndexError(Error):
         def chunk_decref(id, stats):
             nonlocal error
             try:
+                # note: this might do an async repo.delete(id)
                 self.cache.chunk_decref(id, stats)
             except KeyError:
                 cid = bin_to_hex(id)
                 raise ChunksIndexError(cid)
-            except Repository.ObjectNotFound as e:
-                # object not in repo - strange, but we wanted to delete it anyway.
-                if forced == 0:
-                    raise
-                error = True
```

Comment:
this now never happens as all async exceptions get collected and raised with barrier() calls.

previously, this maybe was not very reliable: some exceptions triggered here, some maybe in later/other calls.